Pod/Classes/WPMediaPickerViewController.m

```diff
@@ -945,7 +923,7 @@ - (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSe
 - (void)toggleEmptyViewFor:(NSInteger)numberOfAssets
 {
     if (numberOfAssets > 0) {
-        [self removeContainerEmptyView];
+        self.emptyViewContainer.hidden = true;
     } else {
         [self addContainerEmptyView];
```

Comment:
What about moving this to `viewDidLoad`, and here only hide or show `emptyViewContainer`?

Reply:
I did some tests locally here, and I have some discoveries: If the `parentView` is providing an `emptyViewController` after a search made by the user, it calls the [self.collectionView reloadData]` removing the `self.emptyViewController.view` from subviews. 

So, even calling the `[self.emptyViewContainer setHidden: YES | NO]` shows only the container without subviews. It happens because the `view` comes from an `addChildViewController`, and calling `didMoveToParentViewController`, the view is removed from subviews for some reason, maybe after the `collectionView` reloads the data.

Take a look on the moments:
<img width="1326" alt="Screen Shot 2022-11-03 at 15 19 56" src="https://user-images.githubusercontent.com/20734091/199814218-62e436e2-dff5-4706-9955-5beefd896f81.png">
<img width="1281" alt="Screen Shot 2022-11-03 at 15 20 24" src="https://user-images.githubusercontent.com/20734091/199814271-6dd3b7d8-2876-4c0d-90fc-3e2f09a6c83a.png">

I searched on the codeBase for codes using the `didMoveToParentViewController` specific for the `noResultsViewController`, and it seems all of them remove it from `superview`, instead hide.

It works if we move the `[self populateEmptyViewContainer]` to:
```
- (void)toggleEmptyViewFor:(NSInteger)numberOfAssets
{
    if (numberOfAssets > 0) {
        self.emptyViewContainer.hidden = true;
    } else {
        self.emptyViewContainer.hidden = false;
        [self populateEmptyViewContainer];
    }
}
```
With this approach, we can leave the `[self addEmptyContainerView]` on `viewDidLoad`, but needs to call the populate in the toggle moment. What do you think about that?

I think this was why the older behavior was using `remove/add` the `emptyViewController` at this moment instead `show/hide` logic.

Reply:
I did some tests locally here, and I have some discoveries: 
The `parentView (GutenbergViewController)` provides an `emptyView (NoResultsViewController)`. 
When the `WPMediaPickerViewController` starts the lifecycle and advises the `parentView` from a callback `(didUpdateSearchWithAssetCount:)` that should hide the `emptyView`, in turn, `NoResultsViewController` removes its view from the superView at that moment.

Take a look at the moments:
<img width="1326" alt="Screen Shot 2022-11-03 at 15 19 56" src="https://user-images.githubusercontent.com/20734091/199814218-62e436e2-dff5-4706-9955-5beefd896f81.png">
<img width="766" alt="Screen Shot 2022-11-04 at 16 05 47" src="https://user-images.githubusercontent.com/20734091/200056208-5b90f138-ff0b-40ba-a3f5-bfb7f5af20c2.png">
<img width="671" alt="Screen Shot 2022-11-04 at 16 06 08" src="https://user-images.githubusercontent.com/20734091/200056331-59b067aa-b213-4eed-8da8-54b2fdc2ce46.png">
<img width="1237" alt="Screen Shot 2022-11-04 at 16 06 49" src="https://user-images.githubusercontent.com/20734091/200056355-36ee7193-58d1-45cb-83cc-cc6b229c558d.png">

So, even using hide and show is not enough for our case; Since this `NoResultsViewController` is used in other parts of the App, I don't think it is an option to change the logic in there to don't cause some side effects in other areas.

We can follow this approach, move the `[self populateEmptyViewContainer]` to:
```
- (void)toggleEmptyViewFor:(NSInteger)numberOfAssets
{
    if (numberOfAssets > 0) {
        self.emptyViewContainer.hidden = true;
    } else {
        self.emptyViewContainer.hidden = false;
        [self populateEmptyViewContainer];
    }
}
```
With this, we can leave the `[self addEmptyContainerView]` on `viewDidLoad`, but it needs to call the populate in the toggle moment. What do you think about that?

Reply:
Thanks for the deep dive @diogo-lins! I think your recommendation makes sense 👌